enhancements/network/bgp-overview.md

```diff
@@ -0,0 +1,152 @@
+context and should continue to update this document as necessary.
+
+## Use Cases
+
+### L3 (Layer 3) Network Fabric
+
+The use case here does not require any changes to OpenShift.  It is still
+useful to call out as it is one use of BGP that is relevant to OpenShift,
+particularly with on premise clusters.
+
+It is possible to use BGP between the routers used to build the L3 network that
+OpenShift Nodes are connected to.
+
+### Dynamic Routing for Egress Traffic
+
+Some networks may have more than one router accessible from Nodes for
+availability reasons.  In this case an administrator may want to use BGP down
+to the Nodes themselves so they are aware of the multiple routing options and
+potentially use ECMP (Equal Cost Multi Path) to distribute traffic
+across each of the routers.
+
+This capability is not provided by OpenShift today.  It may be possible as a
+custom configuration, but not with all network providers.  With OpenShift-SDN,
+it should be possible to run your own BGP routing daemon on the host via a
+`DaemonSet`.
+
+OVN-Kuberenetes uses a different host network configuration where egress
+traffic is routed by Open vSwitch flows programmed by OVN.  This mode is
+referred to as "shared gateway mode".  It means that a dynamic routing daemon
+must also feed those routes into OVN configuration, so it's not as straight
+forward as just running an existing daemon on the host.
+
+A future enhancement is needed to describe how we would provide this
+functionality natively with OpenShift.
+
+### Pod Network Control Plane
+
+Many people become aware of BGP in the Kubernetes community via its use in
+[Calico][1].  The primary way BGP is used in Calico is as a control plane for
+the Pod network.  In other words, Calico uses BGP internally for Nodes to share
+the locations of Pod IP addresses in the cluster.
+
+You may read Calico's own documentation to understand more details about how
+Calico uses BGP.
+
+There are other ways this control plane functionality can be implemented, such
+as using the Kubernetes API or by using some other custom control plane
+technology.  Both OpenShift-SDN and OVN-Kubernetes use alternative methods to
+implement their control planes and it is not feasible to change them to use
+BGP for this purpose.
+
+### External Service Load Balancing
+
+Kubernetes Services with `type=LoadBalancer` are typically used as an
+abstraction in front of a cloud's load balancer service.  Many on premise
+cluster environments do not have a load balancer service available, so these
+Services may not be used.
+
+It is possible to implement `LoadBalancer` Services for these on premise
+clusters other ways.  One way involves using a BGP speaker on Nodes to publish
+routes to Nodes where `LoadBalancer` IP addresses may be reached.  [MetalLB][2]
+is one project that implements this technique.  [OpenShift Enhancement #356][3]
+discusses the use of MetalLB with OpenShift in more detail.
```

Comment:
Probably want to mention Service.spec.externalIPs here somewhere.

Reply:
Good point.  I'll add something about externalIPs, and then describe integrated BGP as one way we could provide an automated upgrade to the same use case.